src/repr/src/row.rs

```diff
@@ -1042,6 +1044,17 @@ impl Row {
         }
     }
 
+    /// Unpack `self` into an `OwnedDatums` for efficient random access. This method consumes the
+    /// row and in return allows the caller to keep the unpacked row around.
```

Comment:
We might want to note that this creates one or two heap allocations, which can be costly. It'll create two heap allocations when the row is not spilled, and only one for unpacking the row otherwise.